### PR TITLE
Amplify based React example based on the existing Javascript example

### PR DESCRIPTION
**Pull request by dave-malone**
*Issue #, if available:*

*Description of changes:*

Amplify based React example based on the existing Javascript example that demonstrates how to use Amplify, React, and Cognito to allow users to access video streams using identity managed in Cognito. Includes instructions for how to configure this example project in the user's own AWS account using the Amplify and AWS CLIs. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/amplify-react/README.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+
+An example project for KVS WebRTC. This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app) and was built to work with [Amplify](https://docs.amplify.aws/). AWS services that this project integrates with are Cognito for user authentication and authorization, and Amazon Kinesis Video Streams for WebRTC.
+
```
Comment (unicornss) [inline examples/amplify-react/README.md:3]: link to setup Cognito https://docs.aws.amazon.com/cognito/latest/developerguide/what-is-amazon-cognito.html  ?

**Review by unicornss — COMMENTED** [review]
are there any UI changes that we need to capture as  screenshots for easier reference?

**Comment by disa6302** [comment]
@dave-malone , 

Thank you for your PR. Can we move this to the [demos](https://github.com/aws-samples/amazon-kinesis-video-streams-demos/tree/iot-jenkins/webrtc-js) repository?

File: examples/amplify-react/src/master.js [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    const resolution = widescreen ? { width: { ideal: 1280 }, height: { ideal: 720 } } : { width: { ideal: 640 }, height: { ideal: 480 } };
+    const constraints = {
+      video: sendVideo ? resolution : false,
+      audio: sendAudio,
+    };
+
+    // Get a stream from the webcam and display it in the local view.
+    // If no video/audio needed, no need to request for the sources.
+    // Otherwise, the browser will throw an error saying that either video or audio has to be enabled.
+    if (sendVideo || sendAudio) {
```
Comment (disa6302) [inline examples/amplify-react/src/master.js:87]: To confirm, this means the JS page will allow a `recvonly` option?

**Comment by dave-malone** [comment]
@disa6302 that works for me.

File: examples/amplify-react/src/master.js [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        if (openDataChannel) {
+            this.dataChannelByClientId[remoteClientId] = peerConnection.createDataChannel('kvsDataChannel');
+            peerConnection.ondatachannel = event => {
+                event.channel.onmessage = this.onRemoteDataMessage;
+            };
+        }
+
+        // Poll for connection stats
+        if (!this.peerConnectionStatsInterval) {
+            this.peerConnectionStatsInterval = setInterval(() => peerConnection.getStats().then(this.onStatsReport), 1000);
```
Comment (disa6302) [inline examples/amplify-react/src/master.js:116]: Wondering if 1 second would generate a lot of spew? Can we raise to say 5 seconds?

File: examples/amplify-react/src/viewer.js [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+    this.peerConnection = new RTCPeerConnection(configuration);
+    if (openDataChannel) {
+      this.dataChannel = this.peerConnection.createDataChannel('kvsDataChannel');
+      this.peerConnection.ondatachannel = event => {
+        event.channel.onmessage = this.onRemoteDataMessage;
+      };
+    }
+
+    // Poll for connection stats
+    this.peerConnectionStatsInterval = setInterval(() => this.peerConnection.getStats().then(this.onStatsReport), 1000);
```
Comment (disa6302) [inline examples/amplify-react/src/viewer.js:109]: Same here. Can we raise it to a 5 second default? Why arent we accepting `peerConnectionStatsInterval` setting here like in master and just defaulting?

File: examples/amplify-react/src/App.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    window.addEventListener('beforeunload', this.stopButtonHandler)
+
+    window.addEventListener('unhandledrejection', e => {
+        console.error(e.reason.toString());
+        e.preventDefault();
+    })
+
+    window.addEventListener('error', e => {
+        console.error(e.message);
+        e.preventDefault();
```
Comment (niyatim23) [inline examples/amplify-react/src/App.js:39]: Is `e.preventDefault` to escape unhandled exceptions and compile-time errors when the page loads?

File: examples/amplify-react/src/master.js (hunk 2 above)
Comment (dave-malone) [reply examples/amplify-react/src/master.js:87]: All of my code is simply a refactoring of the code here: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-js/tree/master/examples

File: examples/amplify-react/src/master.js (hunk 3 above)
Comment (dave-malone) [reply examples/amplify-react/src/master.js:116]: This code was taken from the existing example, here: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-js/blob/master/examples/master.js#L136-L139

Whatever value you suggest be implemented here should be implemented in both places.

File: examples/amplify-react/src/App.js (hunk 5 above)
Comment (dave-malone) [reply examples/amplify-react/src/App.js:39]: Sample code incorporated from here: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-js/blob/e6d3be4751d7afadd5e43780aee76c3fa9d096ab/examples/app.js#L94-L97

In react, I'd prefer not to use `window.addEventListener` at all, but this was added to maintain existing functionality

**Pull request closed without merging**